tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java

```diff
@@ -270,6 +270,25 @@ public void listBucket() throws Exception {
         .runAndCheckResult(expected);
   }
 
+  @Test
+  public void listNonExistentBucket() throws Exception {
+    String bucketName = "bucket";
+    //empty parameters
+    List<URIStatus> statuses = mFileSystem.listStatus(new AlluxioURI("/"),
+        ListStatusPOptions.newBuilder().setRecursive(true).build());
+
+    // Verify 404 HTTP status & NoSuchBucket S3 error code
+    HttpURLConnection connection = new TestCase(mHostname, mPort, mBaseUri,
+        bucketName, NO_PARAMS, HttpMethod.GET,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE))
+        .execute();
+    Assert.assertEquals(404, connection.getResponseCode());
+    S3Error response =
+        new XmlMapper().readerFor(S3Error.class).readValue(connection.getErrorStream());
+    Assert.assertEquals(bucketName, response.getResource());
+    Assert.assertEquals(S3ErrorCode.Name.NO_SUCH_BUCKET, response.getCode());
+  }
+
```

Comment:
This test shows that we are stille returning the `NoSuchBucket` S3 error code & HTTP 404 response status after the changes.